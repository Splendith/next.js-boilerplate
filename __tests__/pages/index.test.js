/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import App, { Index } from '~/pages/index';

describe('With Enzyme', () => {
  it('should shows "Hello! Hover me~"', () => {
    const app = shallow(<Index />);

    expect(app
      .find('Header')
      .children()
      .text()).toEqual('Hello! Hover me~');
  });
});

describe('With Snapshot Testing', () => {
  it('shallow snapshot', () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshot = shallowRenderer.render(<Index />);
    expect(snapshot).toMatchSnapshot();
  });
  it('full snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
