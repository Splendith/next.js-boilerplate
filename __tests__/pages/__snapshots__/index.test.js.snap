// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing full snapshot 1`] = `
<div
  className="root"
>
  <div
    className="Wrapper-s17x8e2k-0 jhHFlr"
  >
    <div
      className="Wrapper-s1taabva-0 iVPNZX"
    >
      <div
        className="Box-s1taabva-1 eGPsRV"
      >
        <h1
          className="Title-jn44e3-0 iTtVtA"
        >
          Hello world!
        </h1>
        <div
          className="Image-jn44e3-1 fTtOUh"
        >
          <img
            alt="Smile"
            src="test-file-stub"
          />
        </div>
        <a
          href="/test/5?queryId=8"
          onClick={[Function]}
        >
          Test link with /test/5?queryId=8
        </a>
      </div>
      <div
        className="Box-s168kvmw-0 jSleo"
      >
        <h5>
          Redux global state
        </h5>
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Add count
        </button>
         
        <button
          className="btn btn-outline-secondary"
          onClick={[Function]}
          type="button"
        >
          Clear count
        </button>
        <br />
        <br />
        <div>
          Counter: 
          0
        </div>
        <div>
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing shallow snapshot 1`] = `
<Layout>
  <CounterPage>
    <SideEffect(Head)>
      <title>
        Hello world!
      </title>
    </SideEffect(Head)>
    <Title>
      Hello world!
    </Title>
    <Image>
      <img
        alt="Smile"
        src="test-file-stub"
      />
    </Image>
    <LinkRoutes
      route="/test/5?queryId=8"
    >
      <a>
        Test link with /test/5?queryId=8
      </a>
    </LinkRoutes>
  </CounterPage>
</Layout>
`;
